docs/DeveloperGuide.adoc

```diff
@@ -475,23 +475,36 @@ is rather high. Implementing alternative 2 would be more tedious on the user's s
+The following sequence diagram, Figure 3.5.2, shows further details of the interaction between `ModelManager` and 'ReadOnlyAddressBook` to obtain the blacklist:
 
-`Logic` interface uses `LogicManager`, the class that implements its methods, to package it as a `Command` object. The respective command is then executed by this following statement in `LogicManager` class:
+image::BlacklistCommandSequenceDiagram2.png[width="800"]
+_Figure 3.5.2 : Sequence Diagram of Model and Addressbook Interacts To Generate The Blacklist From All Persons_
+
+The List mechanism is facilitated by commands which use `Logic` interface to obtain the copy of the list that is required. As seen from Figure 3.5.1 and Figure 3.5.2 above, to obtain the current blacklist of the addressbook, "blacklist" `String` is first captured by the `CommandBox` class. The `CommandBox` class then passes this `String` to the `Logic` interface for execution.
+
+`Logic` interface uses `LogicManager` class to firstly validate the written command and package it as a `Command` object. The respective command is then executed in `LogicManager` class. These are the instructions that are executed for this command:
 
 [source,java]
 ----
-CommandResult result = command.execute();
+public CommandResult execute() {
+        model.changeListTo(COMMAND_WORD);
+        model.updateFilteredBlacklistedPersonList(PREDICATE_SHOW_ALL_BLACKLISTED_PERSONS);
+        return new CommandResult(MESSAGE_SUCCESS);
+}
 ----
 
-Specifically for `blacklist` command, it calls the `getFiltererdBlacklistedPersons()` method residing in the `Logic` interface. `Logic` `Manager` subsequently calls `Model` interface. `ModelManager` handles the command and thereafter calls `ReadOnlyAddressBook` interface to handle the request. `AddressBook` class handles this request and thus will return the required list, since the various lists are stored in this class. Technically, the `List` `Object` that exists in the current structure just contains all persons. The blacklisted persons are obtained by running a check on all persons residing in this variable. This is shown by the following code below:
+Specifically for `blacklist` command, it calls the `getFiltererdBlacklistedPersons()` method residing in the `Logic` interface. `LogicManager` subsequently calls `Model` interface. `Model` uses `ModelManager` class to handle the command and thereafter calls `ReadOnlyAddressBook` interface to handle the request. 'ReadOnlyAddressBook` uses `AddressBook` class to handle this request.
 
+In the `AddressBook` class, there is only one `persons` variable that stores all `ReadOnlyPerson` class objects. The blacklisted persons are obtained by running a check on all debtors residing in this variable. The check is executed using the `getIsBlacklisted()` method that resides in the `ReadOnlyPerson` class. This is shown by the following code below:
```

Comment:
...executed using the `ReadOnlyPerson#getIsBlacklisted()` method, as shown below:

Reply:
Noted.